src/client/src/App/Credit/CreditRfqs/CreditRfqCards/Quote/CardQuote.tsx

```diff
@@ -38,9 +43,62 @@ export const Quote = ({
   direction: Direction
   highlight: boolean
 }) => {
-  const priced = quote?.price !== undefined
+  //I would like to consolidate these functions
+  const isQuotePriced = (quote: QuoteBody) => {
```

Comment:
You could move all these functions out of the component to neaten it up a bit if you wanted to.
A pattern I like is array.includes, for example.

```
const pricedQuoteStates = [
  PENDING_WITH_PRICE_QUOTE_STATE,
  ACCEPTED_QUOTE_STATE,
  REJECTED_WITH_PRICE_QUOTE_STATE,
]

const stateType = quote?.state.type ? quote?.state.type : ""
const priced = pricedQuoteStates.includes(stateType)
```
Then it's very quick to add or remove types to that array in the future. That's an option if you like it too.

Reply:
Wow Thank you for this! Will definitely be implementing it this way it is much more concise.

Reply:
Yeah, and when you do use switches, you can rely on follow through for similar cases:

```Typescript
switch (quote.state.type) {
      case PENDING_WITH_PRICE_QUOTE_STATE:
      case ACCEPTED_QUOTE_STATE:
      case REJECTED_WITH_PRICE_QUOTE_STATE:
        return true
      default:
        return false
    }
```